### PR TITLE
Added Jaeger tracing code to exchange app

### PR DESCRIPTION
**Pull request by jramcast**
Tracing code was added for:

- exchange service
- currencies service
- history service

I give priority to configuration using env variables to reduce the initialization code as much as possible.

I also added a template provided by @someth2say that replaces obsolete kubefiles, as well as a readme file to make things easier to anyone that comes across the code.

### CODE REVIEW

**Comment by m-czernek** [comment]
@rsriniva will you want to push this code to a different directory? Different branch?

File: mczernek-exchange-application/kubefiles/exchange-app-template.yml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      hosts:
+      - "*"
+      gateways:
+      - frontend-gw
+      http:
+      - match:
+        - uri:
+            prefix: /frontend
+        rewrite:
+          uri: /frontend
```
Comment (m-czernek) [inline mczernek-exchange-application/kubefiles/exchange-app-template.yml:36]: I've just tested that this is not necessary. I assumed that the prefix will be lost after re-routing, but it is kept. So the rewrite and uri lines can be deleted.

Comment (jramcast) [reply mczernek-exchange-application/kubefiles/exchange-app-template.yml]: Thanks Marek. Fixed!

**Comment by rsriniva** [comment]
most probably i will copy this to a separate folder and make some changes to adapt the code to ch3 EoC lab

**Comment by jramcast** [comment]
@rsriniva @m-czernek   Should I close the PR for now? I can leave the branch as it is so that you can grab the code.

**Comment by stencell** [comment]
@m-czernek @rsriniva @jramcast Can one of you please close this and clean up the branch if it isn't needed or isn't expected to be needed in the future?

**Comment by stencell** [comment]
@m-czernek @rsriniva @jramcast ping on this. please merge or clean this up if it is no longer needed?

**Comment by jramcast** [comment]
I'm closing this.

IIRC I created this PR as an example for Ravi. He integrated this example into his own branch, which he later merged into the course.

**Pull request closed without merging**